wallet2/db/db.go

```diff
@@ -0,0 +1,549 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	//nolint:all
+	_ "github.com/glebarez/go-sqlite"
+)
+
+const (
+	AddressTable     = "addresses"
+	TransactionTable = "transactions"
+	PairTable        = "pairs"
+)
+
+type DB interface {
+	CreateTables() error
+
+	InsertIntoAddress(addr *Address) (*Address, error)
+	InsertIntoTransaction(t *Transaction) (*Transaction, error)
+	InsertIntoPair(key string, value string) (*Pair, error)
+
+	UpdateAddressLabel(addr *Address) (*Address, error)
+
+	GetAddressByID(id int) (*Address, error)
+	GetAddressByAddress(address string) (*Address, error)
+	GetAddressByPath(p string) (*Address, error)
+
+	GetTransactionByID(id int) (*Transaction, error)
+	GetTransactionByTxID(id string) (*Transaction, error)
+
+	GetPairByKey(key string) (*Pair, error)
+	GetTotalRecords(tableName string, query string, args ...any) (int64, error)
+
+	GetAllAddresses() ([]Address, error)
+	GetAllAddressesWithTotalRecords(pageIndex, pageSize int) ([]Address, int64, error)
+
+	GetAllTransactions(query string, args ...any) ([]Transaction, error)
+	GetAllTransactionsWithTotalRecords(pageIndex, pageSize int, query string, args ...any) ([]Transaction, int64, error)
+}
+
+type db struct {
+	*sql.DB
+	context.Context
+}
+
+type Address struct {
+	ID         int       `json:"id"`          // id of wallet
+	Address    string    `json:"address"`     // Address in the wallet
+	PublicKey  string    `json:"public_key"`  // Public key associated with the address
+	Label      string    `json:"label"`       // Label for the address
+	Path       string    `json:"path"`        // Path for the address
+	IsImported bool      `json:"is_imported"` // imported for purpose
```

Comment:
@amirvalhalla why we need to save "IsImported"?
We can get it from Path. right?

Reply:
Yes, we can. I just added this field for the future to help us in the GUI to display imported paths faster.

Reply:
Is this the only reason?

Reply:
Anyway we can construct `AddressPath` once we reading from Database, then we can always check path for this purpose.